dist/react-select.js

```diff
@@ -99,94 +207,109 @@ var Select = React.createClass({
    */
-			options: this.props.options,
 			isFocused: false,
-			isOpen: false,
 			isLoading: false,
+			isOpen: false,
+			options: this.props.options,
 			isReadOnly: this.props.listReadOnlyMode
 		};
 	},
 
 	componentWillMount: function componentWillMount() {
+		var _this = this;
+
 		this._optionsCache = {};
 		this._optionsFilterString = '';
-		this.setState(this.getStateFromValue(this.props.value));
-
-		if (this.props.asyncOptions && this.props.autoload) {
-			this.setState({
-				isLoading: true
-			});
-			this.autoloadAsyncOptions();
-		}
-
-		var self = this;
 		this._closeMenuIfClickedOutside = function (event) {
-			if (!self.state.isOpen) {
+			if (!_this.state.isOpen) {
 				return;
 			}
-			var menuElem = self.refs.selectMenuContainer.getDOMNode();
-			var controlElem = self.refs.control.getDOMNode();
+			var menuElem = ReactDOM.findDOMNode(_this.refs.selectMenuContainer);
+			var controlElem = ReactDOM.findDOMNode(_this.refs.control);
 
-			var eventOccuredOutsideMenu = self.clickedOutsideElement(menuElem, event);
-			var eventOccuredOutsideControl = self.clickedOutsideElement(controlElem, event);
+			var eventOccuredOutsideMenu = _this.clickedOutsideElement(menuElem, event);
+			var eventOccuredOutsideControl = _this.clickedOutsideElement(controlElem, event);
 
 			// Hide dropdown menu if click occurred outside of menu
 			if (eventOccuredOutsideMenu && eventOccuredOutsideControl) {
-				self.setState({
+				_this.setState({
 					isOpen: false
-				}, self._unbindCloseMenuIfClickedOutside);
+				}, _this._unbindCloseMenuIfClickedOutside);
 			}
 		};
-
 		this._bindCloseMenuIfClickedOutside = function () {
-			document.addEventListener('click', self._closeMenuIfClickedOutside);
+			if (!document.addEventListener && document.attachEvent) {
+				document.attachEvent('onclick', _this._closeMenuIfClickedOutside);
+			} else {
+				document.addEventListener('click', _this._closeMenuIfClickedOutside);
+			}
 		};
-
 		this._unbindCloseMenuIfClickedOutside = function () {
-			document.removeEventListener('click', self._closeMenuIfClickedOutside);
+			if (!document.removeEventListener && document.detachEvent) {
+				document.detachEvent('onclick', _this._closeMenuIfClickedOutside);
+			} else {
+				document.removeEventListener('click', _this._closeMenuIfClickedOutside);
+			}
 		};
+		this.setState(this.getStateFromValue(this.props.value));
+	},
+
+	componentDidMount: function componentDidMount() {
+		if (this.props.asyncOptions && this.props.autoload) {
+			this.autoloadAsyncOptions();
+		}
 	},
 
 	componentWillUnmount: function componentWillUnmount() {
 		clearTimeout(this._blurTimeout);
 		clearTimeout(this._focusTimeout);
-
 		if (this.state.isOpen) {
 			this._unbindCloseMenuIfClickedOutside();
 		}
 	},
 
 	componentWillReceiveProps: function componentWillReceiveProps(newProps) {
+		var _this2 = this;
+
+		var optionsChanged = false;
 		if (JSON.stringify(newProps.options) !== JSON.stringify(this.props.options)) {
```

Comment:
@RonanAscendant is the latest upstream change too disruptive? too many conflicts or functionality changes? Otherwise that should be our first option. Plan B is doing what @tkreis does in his fork.